core/src/main/scala/io/projectglow/sql/dsl/package.scala

```diff
@@ -24,15 +24,25 @@ package object dsl {
 
   trait ImplicitOperators {
     def expr: Expression
+
+    // Ensure that lambda variables have unique names for nested functions
+    private var lambdaCounter = 0
+    private def nextLambdaName(): String = {
```

Comment:
How does this change things? Do we have tests?

Reply:
We couldn't have nested higher order functions before. So the new implementation of `mean_substitute` failed analysis. I don't think we need a particular test here.